### ISSUE TITLE
Is 8G GPU enough for pytorch training?

### ISSUE THREAD

**Issue by Haifengtao** [voxelmorph/voxelmorph#220]
**Task** (what are you trying to do/register?)

[please describe task here]

**What have you tried**

Please describe specifics of your approach // use of `vxm`

**Details of experiments**

Please carefully specify details about your experiments. If you are training, when what is the setup? What loss are you using? What does the convergence look like? If you are registering, please show example inputs and outputs. etc.

**Comment by adalca** [comment]
yes. We did most of our training one a single GPU

**Comment by klfxmu** [comment]
> **Task** (what are you trying to do/register?)
> 
> [please describe task here]
> 
> **What have you tried**
> 
> Please describe specifics of your approach // use of `vxm`
> 
> **Details of experiments**
> 
> Please carefully specify details about your experiments. If you are training, when what is the setup? What loss are you using? What does the convergence look like? If you are registering, please show example inputs and outputs. etc.

How can i use gpu?how to setup my code ?

**Comment by adalca** [comment]
@klfxmu sorry but you need to provide waay more information here. What have you tried; have you followed the README instructions? Did you look at our [tutorial](http://tutorial.voxelmorph.net)?

**Comment by klfxmu** [comment]
thanks for your reply：
        I have tried run code with gpu to train my registration model .but wherever I do ,the code just run with cpu. can you tell me how to set my gpu in my linux.




On 02/23/2021 23:28, Adrian Dalca wrote:

@klfxmu sorry but you need to provide waay more information here. What have you tried; have you followed the README instructions? Did you look at our tutorial?

**Comment by adalca** [comment]
Unfortunately this is likely beyond a voxelmorph issue -- are you running keras or pytorch? Either way I would refer you to those forums unless you have specific issues as a result of voxelmorph functionality?

**Comment by klfxmu** [comment]
Thanks for your reply:
            this is exactly beyond a voxelmorph issure.
 I have  followed the README instructions.But the code just run with cpu so that I can not reenact your work.Can you send me the true running instruction in terminal.
thanks for your help sincerely!


| |
柯凌志
|
|
klzxmu@163.com
|
签名由网易邮箱大师定制
On 2/24/2021 02:12，Adrian Dalca<notifications@github.com> wrote：

Unfortunately this is likely beyond a voxelmorph issue -- are you running keras or pytorch? Either way I would refer you to those forums unless you have specific issues as a result of voxelmorph functionality?

**Comment by klfxmu** [comment]
I'm sorry Iforgot to say that I am running Keras .But I have tried pytorch too

**Comment by adalca** [comment]
@klfxmu the instructions we have in the readme won't determine if your code uses cpu or gpu, that's up to the platform you use (e.g. keras) and the installation you have. For most people with tensorflow-gpu installed, those instructions will run voxelmorph on the gpu. 

This probably has to do with your tensorflow/keras installation. So I would recommend reading into this. e.g.:

https://stackoverflow.com/questions/44829085/tensorflow-not-running-on-gpu

https://stackoverflow.com/questions/64467035/tensorflow-uses-cpu-instead-of-gpu

https://stackoverflow.com/questions/63016659/why-tensorflow-not-running-on-gpu-while-gpu-devices-are-identified-in-python

**Comment by klfxmu** [comment]
You're the most patient writer I've ever met。I'll go over the information you gave me right away
Thank you




On 02/24/2021 21:42, Adrian Dalca wrote:

@klfxmu the instructions we have in the readme won't determine if your code uses cpu or gpu, that's up to the platform you use (e.g. keras) and the installation you have. For most people with tensorflow-gpu installed, those instructions will run voxelmorph on the gpu.

This probably has to do with your tensorflow/keras installation. So I would recommend reading into this. e.g.:

https://stackoverflow.com/questions/44829085/tensorflow-not-running-on-gpu

https://stackoverflow.com/questions/64467035/tensorflow-uses-cpu-instead-of-gpu

https://stackoverflow.com/questions/63016659/why-tensorflow-not-running-on-gpu-while-gpu-devices-are-identified-in-python
